src/components/SafeRedirect.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { InjectAppServices } from '../services/pure-di';
+import RedirectToExternalUrl from './RedirectToExternalUrl';
+import RedirectToLogin from './RedirectToLogin';
+
+/**
+ * @param { Object } props
+ * @param { string } props.to
+ * @param { import('../services/pure-di').AppServices } props.dependencies
+ */
+function SafeRedirect({ to, whitelist, dependencies }) {
+  const loginWhitelist =
+    [
+      'https://academy.fromdoppler.com/',
+      'http://prod.doppleracademy.com/',
+      'https://doppleracademy.com/',
+    ] || whitelist;
+
+  const isPartialUrl = (url) => {
+    return RegExp(/^\/[\/\w*]*[\w\?\&\=]*$/).test(url); // eslint-disable-line
+  };
```

Comment:
What do you think about simply:

```js
const isPartialUrl = /^\/[\/\w*]*[\w\?\&\=]*$/.test;
```

Reply:
Why do you need an eslint exception?

Reply:
I am not sure about these criteria, what happens if the redirect destination starts with `?` or has any special character? I think that we can discuss it better in a quick call.

Reply:
> Why do you need an eslint exception?

![image](https://user-images.githubusercontent.com/2439363/71415747-04e17880-263c-11ea-93dd-0ec66dd5d96d.png)

Reply:
And is it wrong?

Reply:
I don't like to have useless warnings... Im taking this characters as characters and not as quantifiers.